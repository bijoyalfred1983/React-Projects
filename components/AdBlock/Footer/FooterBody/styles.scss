@import '../../../theme/variables';

.footerBodyContainer {
  @media #{$medium-and-up} {
    padding: 40px 0;
    display: flex;
    justify-content: space-between;
    margin-left: 20px;
  }
  
  @media #{$small-and-down} {
    margin-top: -20px;
  }
}

.footerBody {
  @media #{$medium-and-up} {
    min-width: 170px;
  }

  @media #{$small-and-down} {
    text-align: center;
  }

  .heading {
    color: $color-light-gray;
    font-size: 18px;
    line-height: 2;
    font-family: $font-az-cond-heavy;

    @media #{$small-and-down} {
      padding-top: 10px;
    }
  }

  .list {
    color: $az-gray-shine;
    font-size: 14px;

    @media #{$small-and-down} {
      padding-bottom: 20px;
    }

    li {
      line-height: 2;
      font-family: $font-az-header;
      
      a {
        color: $az-gray-shine;
      }
    }
  }
}


@media print{
  .footerBodyContainer {
    padding: 40px 0;
    display: flex;
    justify-content: space-between;
    margin-left: 20px;
  }
}


