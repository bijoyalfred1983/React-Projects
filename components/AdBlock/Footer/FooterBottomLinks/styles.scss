@import '../../../theme/variables';

.foooterBottomLinks{
  background-color: $az-light-black;  
  box-shadow: none;

  @media #{$small-and-down} {
    text-align: center;
  }

  ul{
    li{
      display: inline-block;  
      padding-right: 15px;
          
      a{
        line-height: 2;
        color: $color-light-gray;
        font-family: $font-az-header;
        text-decoration: underline;
        font-size: 12px;  
        padding-left: 0;
      }
    }
  }
}
