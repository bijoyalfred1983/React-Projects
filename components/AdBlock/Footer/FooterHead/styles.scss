@import '../../../theme/variables';

.footerHead{
  padding: 40px 0;
}

.footerTop{
  background-color: $az-light-black;
  width: 100%;
  box-shadow: none;

  li{
    display: inline;
    padding-left: 0 !important;    

    @media #{$small-and-down} {
      font-size: 20px;
      font-family: $font-az-cond-heavy;
      background-color: $grey-ultra-dark;
      height: 107px;
      line-height: 5.5;
      margin-bottom: 15px;      

      img{
        width: 60px;
        height: 60px;
        vertical-align: middle;
        margin-left: 10%;
        margin-right: 7%;
      }
    }

    @media #{$medium-and-up} {
      font-family: $font-az-medium;
      font-size: 15px;

      img{
        width: 45px;
        height: 45px;
        vertical-align: middle;
      }
    }
    
    a{
      color: white;
    }

    .rightArrow{
      height: 6px;
      width: 12px;
      margin-left: 5px;
    }
  }  
}

@media print{
  .navBtnPrintWidth{
    width:73% !important;
  }

  .printIconWidth{
    width: 27% !important;
  }

  ul.liWidthPrint li{
    width: 33.33333% !important;
    font-size: 11px !important;
  }

  ul.liWidthPrint li a img{
    width: 30px !important;
    vertical-align: middle;
  }
}
