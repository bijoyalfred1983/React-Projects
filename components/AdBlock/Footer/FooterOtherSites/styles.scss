@import '../../../theme/variables';

.footerOtherContainer{
  padding-bottom: 40px;
}

.footerOther{
  background-color: $az-light-black;
  box-shadow: none;

  .heading{
    color: $color-light-gray;
    font-size: 18px;
    line-height: 2;
    font-family: $font-az-cond-heavy;
    text-align: center;
  }

  .listHolder{
    display: table;
    margin: 0 auto;
  }

  .list{
    color: $az-gray-shine;
    font-size: 14px;    

    li{
      @media #{$medium-and-up} {
        float: left;
      }

      &::after{
        content: '|';      
        padding-right: 12px;        
        padding-left: 12px;
      }

      &:last-child::after{
        content: '';
      }

      a{
        line-height: 2;
        color: $az-gray-shine;
        font-family: $font-az-header;
      }
    }
  }
}

@media print{
  .footerOther .list li{
    float: left;
    padding-right: 20px;
  }
}
