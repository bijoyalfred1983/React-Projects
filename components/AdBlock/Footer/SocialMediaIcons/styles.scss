@import '../../../theme/variables';

.footerTop{
  background-color: $az-light-black;
  width: 100%;
  box-shadow: none;

  @media #{$small-and-down} {
    padding-top: 20px;
    text-align: center;
    padding-bottom: 30px;
  }
  
  @media #{$medium-and-up} {
    text-align: right;
  }

  li{    
    display: inline-block;
    
    a{
      padding: 0 10px;
    }  

    img{
      width: 33px;
      height: 34px;
      vertical-align: middle;
    }
  }
  
  li:last-child{
    @media #{$small-and-down} {
      img{
        width: 101px;
      }
    }

    @media #{$medium-and-up} {
      display: none;
    }   
  } 
}

@media print{
  .footerTop li:last-child{
    display: none;
  }

  .footerTop img{
    width: 30px !important;
  }
}
