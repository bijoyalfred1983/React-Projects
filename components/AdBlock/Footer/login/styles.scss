@import '../../../theme/variables';

.loginSection{
  .leftContent{    
    height: 110px;
    background-color: #f26100;
    color: white;    
    padding-top: 15px;

    @media #{$small-and-down} {  
      text-align: center;
    }

    @media #{$medium-and-up} {  
      float: left;
      width: 50%;
      padding-left: 50px;
    }

    h4{
      font-size: 24px;

      @media #{$medium-and-up} {  
        font-size: 35px;
        font-family: $font-az-cond;
        margin-bottom: 0;
        line-height: 1.14;
      }
    }

    div{
      font-size: 16px;

      @media #{$small-and-down} {
        padding: 0 40px;
        line-height: 20px;
        letter-spacing: 0.2px;
      }
    }
  }

  .loginHolder{
    background-color: #f26100;

    @media #{$medium-and-up} {   
      height: 110px;
      float: left;
      width: 50%;
      position: relative;
    }

    .loginContent{
      @media #{$medium-and-up} {
        position: absolute;
        left: 0;
        bottom: 0;
        right: 0;
        border-bottom: 100px solid #fff;
        border-left: 50px solid transparent;
      }
    }

    .fieldWrapper{
      @media #{$medium-and-up} {
        position: relative;
        top: 95px;
      }

      @media #{$small-and-down} {
        padding: 30px 10px 10px;
      }      
    }
  }
}

@media print{
  .loginHolder{
    height: 110px;
    float: left;
    width: 50%;
    position: relative;
  }

  .leftContent{
    float: left;
    width: 50%;
    padding-left: 50px;
  }

  .loginContent{
    position: absolute;
    left: 0;
    bottom: 0;
    right: 0;
    border-bottom: 100px solid #fff;
    border-left: 50px solid transparent;
  }

  .fieldWrapper{
    position: relative;
    top: 143px;
  }

  .printMediaWidth{
    width: 60.33333% !important;
  }

  .hide-on-med-and-up{
    display:none;
  }
}

