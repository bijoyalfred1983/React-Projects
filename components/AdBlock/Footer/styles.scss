@import '../../theme/variables';

.footerWrapper{
  background-color: $az-light-black;
  clear: both;
  position: relative;

  hr{
    border: 1px solid $az-gray-black;
    background-color: $az-light-black; 
    margin: 0 20px;   
  }
}

.footerMainWrapper{
  background-color: $az-light-black;
  clear: both; 
}

.footerMainContent{
  margin: auto;

  @media #{$medium-and-up} {
    width: 80%;
  }
}

.footerCopyright{
  background-color: $az-light-black;
  color: $color-light-gray;
  font-family: $font-az-header;      
  font-size: 12px;
  padding-bottom: 20px;
  clear: both;

  @media #{$small-and-down} {
    padding-left: 20px; 
    padding-top: 30px;
  }
}

.bottomSection{  
  background-color: $az-light-black;
  padding-top: 30px;

  @media #{$medium-and-up} {
    padding-left: 20px; 
  }
}

.hiring{  
  text-align: center;  
  margin: 0 10px 25px 10px;
  color: $color-light-gray;

  span{
    font-size: 20px;
    font-family: $font-az-cond-heavy;
  }

  img{
    width: 45px;
    height: 45px;
    vertical-align: middle;
    margin-top: -15px;
  }

  a{
    display: block;
    color: white;
    font-size: 15px;
    border: 1px solid white;
    width: 160px;    
    padding: 10px ;
    font-family: $font-az-cond;   
    margin: auto;
    margin-top: 5px;
  }
}

.digitalCert {
  position: absolute;
  right: 20px;
  bottom: 15px;
}

@media print{
  .footerMainContent{
    width: 80%;
  }

  .bottomSection{
    padding-left: 20px;
  }

  .printHide{
    display:none;
  }
}
