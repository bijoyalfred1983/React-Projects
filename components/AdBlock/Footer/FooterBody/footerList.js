/* @flow */

import React from 'react';
import type { Element } from 'react';
import styles from './styles.scss';
import { jsTrack } from '../../../containers/Analytics/ClickTrack';

type Props = { footerColumn: Array<Object> };

export const trackFooterLink = (e, linkName) => {
  jsTrack.trackFooterNavMenu(linkName);
};

const FooterList = ({ footerColumn }: Props): Element<'div'> => (
  <div className={styles.footerBody}>
    <div className={`${footerColumn.className}`}>
      <div className={styles.heading}>{footerColumn.title.toUpperCase()}</div>
      <ul className={styles.list}>
        {footerColumn.list.map(listItem => (
          <li key={listItem.id}>
            <a
              href={listItem.URL}
              onClick={e => trackFooterLink(e, listItem.content)}
            >
              {' '}
              {listItem.content}
            </a>
          </li>
        ))}
      </ul>
      <hr className="hide-on-med-and-up" />
    </div>
  </div>
);

export default FooterList;
