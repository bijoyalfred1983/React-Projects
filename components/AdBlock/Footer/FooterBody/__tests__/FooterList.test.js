/* @flow */
import React from 'react';
import { createStore } from 'redux';
import { Provider, connect } from 'react-redux';
import { MuiThemeProvider } from 'material-ui/styles';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import footerList from '../footerList';
const col1 = {
  id:'1000',
  title: 'Shop',
  className: '',
  list: [
    { id: '1', content: 'Store Locator', URL: '' },
    { id: '2', content: 'Shop by Make', URL: '' },
    { id: '3', content: 'Shop by Model', URL: '' },
    { id: '4', content: 'Vin Decoder', URL: '' },
    { id: '5', content: 'AutoZone Rewards', URL: '' },
    { id: '6', content: 'Gift Cards ', URL: '' }
  ]
};

const col2 = {
  id:'10001',
  title: 'AutoZone Services',
  className: 'hide-on-small-only',
  list: [
    { id: '51', content: 'Buy Online, Pick Up in Store', URL: '' },
    { id: '52', content: 'Local Store Ad', URL: '' },
    { id: '53', content: 'Loan-A-Tool', URL: '' },
    { id: '54', content: 'In-store Services', URL: '' },
    { id: '55', content: 'Video Library', URL: '' },
    { id: '56', content: 'Mobile App', URL: '' }
  ]
};

describe('<footerList />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      data: [col1,col2]
    };
  });

  test('renders on footerList correctly', () => {
    const wrapper = mount(
        <footerList {...PROPS} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


