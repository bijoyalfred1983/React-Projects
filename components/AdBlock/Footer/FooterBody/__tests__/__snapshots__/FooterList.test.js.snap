// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<footerList /> renders on footerList correctly 1`] = `<footerList />`;
